src/main/java/mezz/jei/gui/TooltipRenderer.java

```diff
@@ -1,44 +1,234 @@
+import java.util.stream.Collectors;
 
 public final class TooltipRenderer {
-	private TooltipRenderer() {
-	}
+    private TooltipRenderer() {
+    }
+
+    public static void drawHoveringText(Minecraft minecraft, String textLine, int x, int y) {
+        drawHoveringText(ItemStack.EMPTY, minecraft, Collections.singletonList(textLine), x, y, -1, minecraft.fontRenderer);
+    }
+
+    public static void drawHoveringText(Minecraft minecraft, List<String> textLines, int x, int y) {
+        drawHoveringText(ItemStack.EMPTY, minecraft, textLines, x, y, -1, minecraft.fontRenderer);
+    }
+
+    public static void drawHoveringText(Minecraft minecraft, List<String> textLines, int x, int y, int maxWidth) {
+        drawHoveringText(ItemStack.EMPTY, minecraft, textLines, x, y, maxWidth, minecraft.fontRenderer);
+    }
+
+    public static void drawHoveringText(Minecraft minecraft, List<String> textLines, int x, int y, FontRenderer font) {
+        drawHoveringText(ItemStack.EMPTY, minecraft, textLines, x, y, -1, font);
+    }
+
+    public static void drawHoveringText(Minecraft minecraft, List<String> textLines, int x, int y, int maxWidth, FontRenderer font) {
+        drawHoveringText(ItemStack.EMPTY, minecraft, textLines, x, y, maxWidth, font);
+    }
+
+    public static void drawHoveringText(ItemStack itemStack, Minecraft minecraft, List<String> textLines, int x, int y, FontRenderer font) {
+        drawHoveringText(itemStack, minecraft, textLines, x, y, -1, font);
+    }
+
+    public static void drawHoveringText(ItemStack itemStack, Minecraft minecraft, List<String> textLines, int x, int y, int maxWidth, FontRenderer font) {
+        ScaledResolution scaledresolution = new ScaledResolution(minecraft);
+        GuiUtils.drawHoveringText(itemStack, textLines, x, y, scaledresolution.getScaledWidth(), scaledresolution.getScaledHeight(), maxWidth, font);
+    }
+
+    public static void drawHoveringText(ItemStack itemStack, Minecraft minecraft, List<String> textLines, int x, int y, int maxWidth) {
+        drawHoveringText(itemStack, minecraft, textLines, x, y, maxWidth, minecraft.fontRenderer);
+    }
+
+    public static void drawHoveringTextAndItems(Minecraft minecraft, List<Object> textLines, int x, int y) {
+        drawHoveringTextAndItems(ItemStack.EMPTY, minecraft, textLines, x, y, -1, minecraft.fontRenderer);
+    }
+
+    public static void drawHoveringTextAndItems(ItemStack stack, Minecraft minecraft, List<Object> lines, int mouseX, int mouseY, int maxTextWidth, FontRenderer font) {
+        // Almost a copy from GuiUtils.drawHoveringText, but also allowing IngredientListBatchRenderer lines.
+
+        ScaledResolution scaledresolution = new ScaledResolution(minecraft);
+        int screenWidth = scaledresolution.getScaledWidth();
+        int screenHeight = scaledresolution.getScaledHeight();
+        List<String> textLinesOnly = lines.stream().filter(String.class::isInstance).map(String.class::cast).collect(Collectors.toList());

```

Comment:
Possibly pass list of text and renderer as different arguments? This being called every time text is drawn isn't great.

Reply:
I get you, try to avoid `instanceof` checks and streams wherever possible, but if it really is needed then it's fine, the code just looks a tad unnatural at the moment!